.github/workflows/primer_run_main.yaml

```diff
@@ -0,0 +1,82 @@
+# Most of this is inspired by the mypy primer
+# See: https://github.com/hauntsaninja/mypy_primer
+# This is the primer job that runs on the default 'main' branch
+# It is also responsible for caching the packages to prime on
+
+name: Primer / Main
+
+on:
+  push:
+    branches:
+      - main
+
+env:
+  CACHE_VERSION: 1
+
+jobs:
+  run-primer:
+    name: Run / ${{ matrix.python-version }}
+    runs-on: ubuntu-latest
+    timeout-minutes: 45
+    strategy:
+      matrix:
+        python-version: ["3.10"]
```

Comment:
I think we should use the most popular interpreter here. It look like right now it's 3.8. 

```suggestion
        python-version: ["3.8"]
```

![popular_interpreter](https://user-images.githubusercontent.com/5493666/170654540-cddfe75d-c897-428e-94f7-fa5f959c6f16.png)